### DISCUSSION TITLE
Priority "IMMEDIATE" Features

### DISCUSSION THREAD

**Discussion by ergonyc** [ndcn/omicser#150] [Ideas]
What does the NDCN browser need immediately to become a "daily" tool for our constituent labs?

Some initial features identified in our "kick off" discussion:

- [ ] export figure
- [ ] clarified input ("ingest") / output ("export") conversion 
- [ ] ?

**Comment by ddifraia** [comment]
- clarified input ("ingest") / output ("export") conversion <- for me definitely this points remains the biggest bottleneck. 

Regarding exporting figures, is also very important, but probably more easy to implement.

PS: Just as an idea would be cool to implement a "data curation" procedure inside the omicser itself, so that even not coding proficient people could successfully set up they own data... 

In the end you would just need (theoretically)

- a matrix (feature x sample)
- a sample information (metadata)
- feature information (metadata)
- if present a differential expression table.

The column name could be selected by the user, in the sense that the user itself could specify a bit the format of the different file (es tab separated file, with this column indicating this specific characteristics), but this is in a more GUI-user friendly way in omicser. 

This will allow (in principle) non-coding people to upload they data indipendently.

**Reply by ergonyc** [reply]
Ciao @ddifraia, 

I think it could be done pretty straightforwardly.  Someone just needs to code up the interface to uploading the data (matrix and two tables) and choose parameters for differential expression table computation.

Do you want to make a stab at making a UI which could replace the curation steps 7-10 of the "curate_and_config" scripts?  I could help think through how to best integrate it with the current browser.

Issues:
- UI format:  modal sequence vs. tabs/widgets for the UI
- read/write permissions:  giving the app proper permissions to write the database to the database directories
- exit/entry into the overall omicser?

Is anyone else out there interested in seeing this?

**Comment by ricoderks** [comment]
I think this is a really valuable addition, biggest issue probably how to keep the curation part away from the general users. Maybe it is an idea to make a seperate app who does this? My reason behind this is to keep omicser as clean as possible and keep "administration" things out there.

**Reply by ergonyc** [reply]
Keeping the main purpose of the browser to "visualize" and explore makes sense.  The choice to make the "ingestion" options as modals (pop-ups) was to try and keep the tool uncluttered.    Possibly building out the capabilities / logic of those modals?

**Reply by ricoderks** [reply]
That's a nice idea! With the modals we can add functionality without cluttering the browser too much.
